source/base/parameter_handler.cc

```diff
@@ -1313,10 +1638,141 @@ namespace Patterns
       return std::unique_ptr<DirectoryName>();
   }
 
-}   // end namespace Patterns
 
 
 
+  template<>
+  std::unique_ptr<Patterns::PatternBase> default_pattern<int>()
+  {
+    return std::unique_ptr<Patterns::PatternBase>(Integer().clone());
```

Comment:
too complicated :-) Just say
```
  return std::unique_ptr<Patterns::PatternBase>(new Integer());
```
or even more concisely:
```
  return std_cxx14::make_unique_ptr<Integer>();
```
:-)

Same for the functions below. You can pass constructor arguments for the type you're creating to `make_unique` instead and it passes them on to the constructor of the object it creates.